src/main/java/core/basesyntax/bookstore/service/impl/CartItemServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package core.basesyntax.bookstore.service.impl;
+
+import core.basesyntax.bookstore.dto.item.CartItemDto;
+import core.basesyntax.bookstore.dto.item.CreateCartItemDto;
+import core.basesyntax.bookstore.dto.item.UpdateCartItemDto;
+import core.basesyntax.bookstore.exception.EntityNotFoundException;
+import core.basesyntax.bookstore.mapper.CartItemMapper;
+import core.basesyntax.bookstore.model.CartItem;
+import core.basesyntax.bookstore.model.ShoppingCart;
+import core.basesyntax.bookstore.repository.book.BookRepository;
+import core.basesyntax.bookstore.repository.cart.ShoppingCartRepository;
+import core.basesyntax.bookstore.repository.item.CartItemRepository;
+import core.basesyntax.bookstore.service.CartItemService;
+import core.basesyntax.bookstore.service.UserService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+public class CartItemServiceImpl implements CartItemService {
+    private final UserService userService;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    @Transactional
+    public CartItemDto save(CreateCartItemDto cartItemDto) {

```

Comment:
There are too many responsibilities for one method here, please make this more SOLID